artifact/gem5art/artifact/_filedb.py

```diff
@@ -0,0 +1,90 @@
+import json
+from pathlib import Path
+from typing import Optional, Iterable, List, Dict, Union, Tuple, Any
+from uuid import UUID
+import copy
+
+class FileDB:
+
+    """
+        An artifact database with a json file backend.
+    """
+
+    __json_file: Path
+    __uuid_artifact_map: Dict[UUID, Dict[str,str]]
+    __hash_uuid_map: Dict[str, List[UUID]]
+
+    def __init__(self, json_file: Union[str, Path]) -> None:
+        self.__json_file = Path(json_file)
+        self.__uuid_artifact_map, self.__hash_uuid_map = \
+            self.__load_from_file(self.__json_file)
+
+    def __load_from_file(self, json_file: Path) -> Tuple[Dict[UUID, Dict[str,str]], Dict[str, List[UUID]]]:
```

Comment:
single underscore. Double underscore is usually used for python-specific operators.